packages/compiler/src/core/emitter-utils.ts

```diff
@@ -21,5 +22,10 @@ export async function emitFile(program: Program, options: EmitFileOptions): Prom
     options.newLine && options.newLine === "crlf"
       ? options.content.replace(/(\r\n|\n|\r)/gm, "\r\n")
       : options.content;
+
+  if (program.compilerOptions.listFiles) {
```

Comment:
I think we might want to collect this list and emit it at the end of each emitter instead of when they are emitted

Reply:
that is the current workflow, it is just saving the emitter file name in a list but it is printed until the emitter finish, is it what you mean?